crates/oxc_ast/src/trivia.rs

```diff
@@ -57,6 +57,17 @@ impl Trivias {
         self.comments.range(span.start..span.end).count() > 0
     }
 
+    #[must_use]
+    pub fn get_leading_comment_span(&self, span: Span) -> Option<Span> {
+        let comments = self.comments.range(..span.start);
+        match comments.last() {
+            Some((start, comment)) if comment.end() + 2 == span.start => {
```

Comment:
The idea was that the end of the comment span + 2 would be the start of the node span, but it's not entirely true because the end of the comment span is not actually the closing `/`, since that is trimmed out in `TriviaBuilder::add_multi_line_comment`. In general, it's very volatile and we should not go with this approach – I'd love some guidance on how we can properly attribute a given node's span to a comment span